src/apps/external_executor/external_executor.cpp

```diff
@@ -681,18 +688,81 @@ namespace externalexecutor
               break;
             }
           }
+
+          out_stream->stream_msg(result);
         }
 
-        return ccf::grpc::make_success(results);
+        return ccf::grpc::make_success();
       };
 
       make_command_endpoint(
         "/temp.Test/RunOps",
         HTTP_POST,
-        ccf::grpc_command_adapter<
+        ccf::grpc_command_unary_stream_adapter<
           std::vector<temp::OpIn>,
-          std::vector<temp::OpOut>>(run_string_ops),
+          temp::OpOut>(run_string_ops),
+        ccf::no_auth_required)
+        .install();
+
+      auto sub = [this](
+                   ccf::endpoints::CommandEndpointContext& ctx,
+                   temp::Event&& payload,
+                   ccf::grpc::StreamPtr<temp::EventInfo>&& out_stream) {
+        std::unique_lock<ccf::pal::Mutex> guard(subscribed_events_lock);
+
+        subscribed_events.emplace(std::make_pair(
+          payload.SerializeAsString(),
+          ccf::grpc::detach_stream(std::move(out_stream))));
+        LOG_INFO_FMT("Subscribed to event {}", payload.SerializeAsString());
+
+        return ccf::grpc::make_pending();
+      };
+      make_endpoint(
+        "/temp.Test/Sub",
+        HTTP_POST,
+        ccf::grpc_command_unary_stream_adapter<temp::Event, temp::EventInfo>(
+          sub),
+        {ccf::no_auth_required})
+        .install();
+
+      auto pub = [this](
+                   ccf::endpoints::CommandEndpointContext& ctx,
+                   temp::EventInfo&& payload)
+        -> ccf::grpc::GrpcAdapterResponse<google::protobuf::Empty> {
+        temp::Event event;
+        event.set_name(payload.name());
+
+        std::unique_lock<ccf::pal::Mutex> guard(subscribed_events_lock);
+
+        auto search = subscribed_events.find(event.SerializeAsString());
+        if (search != subscribed_events.end())
+        {
+          if (!search->second->stream_msg(payload))
+          {
+            // Manual cleanup of closed streams. We should have a close
+            // callback for detached streams to cleanup resources when
+            // required instead
```

Comment:
Does that need a ticket?

Reply:
This is item 5. in https://github.com/microsoft/CCF/issues/4325.